app/src/main/java/com/duckduckgo/app/browser/senseofprotection/SenseOfProtectionExperimentImpl.kt

```diff
@@ -0,0 +1,179 @@
+)
+@SingleInstanceIn(AppScope::class)
+class SenseOfProtectionExperimentImpl @Inject constructor(
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+    private val userBrowserProperties: UserBrowserProperties,
+    private val senseOfProtectionToggles: SenseOfProtectionToggles,
+    private val senseOfProtectionPixelsPlugin: SenseOfProtectionPixelsPlugin,
+    private val pixel: Pixel,
+) : SenseOfProtectionExperiment {
+
+    init {
+        // enrol users in the existing user experiment if they are not already enrolled in the new user experiment
+        appCoroutineScope.launch(dispatcherProvider.io()) {
+            if (userBrowserProperties.daysSinceInstalled() > EXISTING_USER_DAY_COUNT_THRESHOLD) {
+                if (!isEnrolledInNewUserExperiment()) {
+                    enrollInExistingUserExperiment(cohortName = MODIFIED_CONTROL)
+                }
+            }
+        }
+    }
+
+    override fun enrolUserInNewExperimentIfEligible(): Boolean {
+        return if (userBrowserProperties.daysSinceInstalled() <= EXISTING_USER_DAY_COUNT_THRESHOLD) {
+            enrollInNewUserExperiment(cohortName = MODIFIED_CONTROL)
+        } else {
+            false
+        }
+    }
+
+    override fun isUserEnrolledInModifiedControlCohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == MODIFIED_CONTROL.cohortName && isNewUserExperimentEnabled(MODIFIED_CONTROL) ||
+            getExistingUserExperimentCohortName() == MODIFIED_CONTROL.cohortName && isExistingUserExperimentEnabled(MODIFIED_CONTROL)
+
+    override fun isUserEnrolledInVariant1CohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == VARIANT_1.cohortName && isNewUserExperimentEnabled(VARIANT_1) ||
+            getExistingUserExperimentCohortName() == VARIANT_1.cohortName && isExistingUserExperimentEnabled(VARIANT_1)
+
+    override fun isUserEnrolledInVariant2CohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == VARIANT_2.cohortName && isNewUserExperimentEnabled(VARIANT_2) ||
+            getExistingUserExperimentCohortName() == VARIANT_2.cohortName && isExistingUserExperimentEnabled(VARIANT_2)
+
+    override fun isMemberOfVariant2Cohort(): Boolean =
+        senseOfProtectionToggles.senseOfProtectionNewUserExperimentApr25()
+            .getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName ||
+            senseOfProtectionToggles.senseOfProtectionExistingUserExperimentApr25()
+            .getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName
+
+    override fun shouldShowNewPrivacyShield(): Boolean {
+        return isUserEnrolledInVariant1CohortAndExperimentEnabled() || isUserEnrolledInVariant2CohortAndExperimentEnabled()
+    }
+
+    override fun getTabManagerPixelParams(): Map<String, String> {
+        return when {
+            isEnrolledInNewUserExperiment() -> {
```

Comment:
This still seems to only check if the user is enrolled rather than is enrolled and enabled so the parameters will be added even when the experiment is disabled which is probably wrong?

Reply:
You are right, like this the parameters are always added.

Reply:
Fixed it and added tests